### ISSUE TITLE
Border

### ISSUE THREAD

**Issue by Logan7432** [Megabyte918/MultiOgar-Edited#672]
How do I fix it so the border fits my map? https://ibb.co/jvfB8v https://ibb.co/fVWeva

**Comment by MastaCoder** [comment]
Make your map smaller? lol

**Comment by Logan7432** [comment]
@MastaCoder Im not sure how to though, like what are all those numbers. https://ibb.co/daHG8v

**Comment by Logan7432** [comment]
@MastaCoder the old one had this https://ibb.co/fbVQ1F

**Comment by Logan7432** [comment]
Please giveme a good awnswer this time

**Comment by Logan7432** [comment]
@MastaCoder @Megabyte918 Can you please help me out, I cannot figure out how to resize my border and fit it to my mini map. Can you please tell me where to look in the code and explain what numbers I need to change. I never ran into this problem on the old Cigar server. I need a reply asap. Thanks.

**Comment by AlexHGaming** [comment]
14142x14142 is the default...

**Comment by Luka967** [comment]
The new configs set the map's full width and height. The map's center is always 0,0.
Would draw a sketch in paint but can't atm xd

**Comment by Logan7432** [comment]
@Luka967 Ok, if you get time please help, but I'm just confused because the old one had like 0 left and then 30000 right and not sure what that would convert to on this version

**Comment by Luka967** [comment]
Set both to 30000.

**Comment by Logan7432** [comment]
@Luka967, Ok, I set it to 30000, but its still way off my mini-map. See for yourself on my site, WilsonAgario.ddns.net

**Comment by Luka967** [comment]
Remember that the center is now at 0,0; the new border in the old config way would be -15000, 15000, -15000 and 15000 respectively.

**Comment by Logan7432** [comment]
@luka967 thanks, this will be very helpful, but just a question would there be a way to change it from 0,0 to -15000? If not that's fine

**Comment by Gigabyte918** [comment]
what is the client you are using @Logan7432

**Comment by MastaCoder** [comment]
Turn off server minimap scramble.

**Comment by Logan7432** [comment]
@MastaCoder There is no mini-map scramble, I never had this problem with the old cigar client. When i try and use the old server it crashes for some reason

**Comment by Logan7432** [comment]
@Gigabyte918 Im using a very very heavly coded https://github.com/dinhduynhat/Agar-Ogar-Cigar-web-client. Btw I was the one who created this client and I shared it with someone to get help and then they claimed it as theirs and posted it which dosnt make me happy.

**Comment by MastaCoder** [comment]
"Heavy Coded" - Uses Comic Sans as website font with a generic attempt to un-obfuscate a minimap from another website.

**Comment by Logan7432** [comment]
@MastaCoder When I first made the site there was nothning but a play button and a crappy chat box, Now the site is very customized to my liking and had a lot of work put into it, so I would call it heavily coded.

**Comment by Logan7432** [comment]
@MastaCoder So is there any way to get the minimap working with the server, cause that is a close copy of the client I use.

**Comment by gyeyoqu** [comment]
You can to adapt your minimap code to accept all sizes of map

**Comment by gyeyoqu** [comment]
@Logan7432

**Comment by Logan7432** [comment]
@gyeyoqu I'll try and figure out how to do that, if you have any mini map code that does that please hook me up

**Comment by Logan7432** [comment]
@Luka967 Where would I go in the server server or client to change the mini-map from -15000, 15000, -15000 and 15000 to 0,0?

**Comment by Logan7432** [comment]
fixed

**Comment by Logan7432** [comment]
Well, i just realized that Multi Ogar has some features I need. I need To somehow get this Map problem fixed.

**Comment by Logan7432** [comment]
https://ibb.co/gou9dv

**Comment by Logan7432** [comment]
@MastaCoder

**Comment by Logan7432** [comment]
This makes no sense, Mostly every other sever works fine, but no..... the good server dosnt work with my minimap..... Theres something wrong. You guys coded it you should know how to fix the border

**Comment by Logan7432** [comment]
The bottem right corner of my map is in the center of the minimap just for some help https://ibb.co/frnsWF

**Comment by MastaCoder** [comment]
change the map coordinates to add half of the map size to the actual map length.

**Comment by Logan7432** [comment]
@MastaCoder Ok, so my map right now si 18000, So should I Make it 9,000? If its possible can you direct me to the line of code?

**Comment by Logan7432** [comment]
@MastaCoder I made it 9,000 and this is what happened https://ibb.co/cOQkka

**Comment by Logan7432** [comment]
@MastaCoder Please try and find a fix for me, I've been trying to fix this for hours and I cant get it to work!

**Comment by Logan7432** [comment]
@MastaCoder I need this fixed ASAP

**Comment by MastaCoder** [comment]
1. this is client sided.
2. the minimap is a stolen minimap from a project, and appears to be an attempt at cleaning up an obfo. minimap from another website. (Which is honestly just considered stealing now)
3. http://codecademy.com
4. http://bfy.tw/AkaV
5. There is no 5 :trollface:

**Comment by Logan7432** [comment]
@MastaCoder How is it Client sided when every other server works fine on it?

**Comment by Megabyte918** [comment]
@Logan7432 It could be an issue with your minimap within the client itself, regular clients/extensions with working minimaps should be able to be adapted to whatever dimensions the map is set to be in gameserver.ini.

**Comment by Logan7432** [comment]
@megabytes918 is there any clients you know of that I could use to test this?

**Comment by Jhair2015** [comment]
@Logan7432 AgarPlus.

**Comment by Logan7432** [comment]
@Jhaie2015 Isn't that a server not a client?

**Comment by Logan7432** [comment]
I've had no success on fixing this issue.

**Comment by Jhair2015** [comment]
No, is a client created since agarplus

**Comment by Logan7432** [comment]
@Jhair2015 Sorry, I dont understand what your saying. Can you provide a link?

**Comment by Logan7432** [comment]
@MastaCoder @Luka967 @Gigabyte918 @gyeyoqu @Megabyte918 What Client do you guys run your server on? If it works for you may I use your client files to get my client working?

**Comment by Logan7432** [comment]
CAN SOMEONE JUST ANSWER WITH A HELPFUL RESPONSE!

**Comment by Logan7432** [comment]
! function() {
	    function i(i, t) {
	        var n = u("<div>").attr("id", "mini-map-token-" + i).css({
	            position: "absolute",
	            width: "5%",
	            height: "5%",
	            background: t,
	            top: "0%",
	            left: "0%"
	        });
	        return n
	    }

	    function t(i, t) {
	        void 0 === window.mini_map_tokens[i] && (window.mini_map.append(t), window.mini_map_tokens[i] = t)
	    }

	    function n(i) {
	        void 0 !== window.mini_map_tokens[i] && (window.mini_map_tokens[i].detach(), delete window.mini_map_tokens[i])
	    }

	    function e(i) {
	        return void 0 !== window.mini_map_tokens[i]
	    }

	    function o(i, t, n) {
	        return void 0 !== window.mini_map_tokens[i] ? (window.mini_map_tokens[i].css("left", t / 11e3 * 90 + "%").css("top", n / 11e3 * 90 + "%"), !0) : !1
	    }

	    function s(i, t) {
	        window.mini_map_pos.text("x: " + i.toFixed(0) + ", y: " + t.toFixed(0))
	    }

	    function a() {
	        var i = window.jQuery;
	        if (window.mini_map_tokens = {}, 0 === i("#mini-map-pos").length && (window.mini_map_pos = i("<div>").attr("id", "mini-map-pos").css({
	                top: 10,
	                left: 160,
	                color: "white",
	                background: "rgba(155, 155, 155, 0.6)",
	                fontSize: 24,
	                fontWeight: 800,
	                position: "fixed",
	                padding: "0px 10px"
	            }).appendTo(document.body)), 0 === i("#mini-map-wrapper").length) {
	            var t = i("<div>").attr("id", "mini-map-wrapper").css({
	                    position: "fixed",
	                    bottom: "110px",
	                    right: "10px",
	                    width: "150px",
	                    height: "150px",
	                    background: ""
	                }),
	                n = i("<div>").attr("id", "mini-map").css({
	                    width: "67%",
	                    height: "67%",
	                    position: "relative"
	                });
	            t.append(n).appendTo(document.body), window.mini_map = n
	        }
	    }

	    function r(i, t, n, e, o, s) {
	        m[i] = this, this.id = i, this.ox = this.x = t, this.oy = this.y = n, this.oSize = this.size = e, this.color = o, this.points = [], this.pointsAcc = [], this.setName(s)
	    }

	    function d(i, t, n) {
	        Object.defineProperty(i, n, {
	            get: function() {
	                return t[n]
	            },
	            set: function(i) {
	                t[n] = i
	            },
	            enumerable: !0,
	            configurable: !0
	        })
	    }

	    function p(i, t) {
	        var n = +new Date,
	            e = Math.random(),
	            o = t,
	            s = i.getUint16(o, !0);
	        o += 2;
	        for (var a = 0; s > a; ++a) {
	            var d = m[i.getUint32(o, !0)],
	                p = m[i.getUint32(o + 4, !0)],
	                o = o + 8;
	            d && p && (p.destroy(), p.ox = p.x, p.oy = p.y, p.oSize = p.size, p.nx = d.x, p.ny = d.y, p.nSize = p.size, p.updateTime = n)
	        }
	        for (a = 0;;) {
	            var c = i.getUint32(o, !0);
	            if (o += 4, 0 == c) break;
	            ++a;
	            var d = i.getInt16(o, !0),
	                o = o + 2,
	                p = i.getInt16(o, !0),
	                o = o + 2;
	            g = i.getInt16(o, !0), o += 2;
	            for (var h = i.getUint8(o++), u = i.getUint8(o++), l = i.getUint8(o++), h = (h << 16 | u << 8 | l).toString(16); 6 > h.length;) h = "0" + h;
	            var h = "#" + h,
	                w = i.getUint8(o++),
	                u = !!(1 & w),
	                l = !!(16 & w);
	            2 & w && (o += 4), 4 & w && (o += 8), 8 & w && (o += 16);
	            for (var f, w = ""; f = i.getUint16(o, !0), o += 2, 0 != f;) w += String.fromCharCode(f);
	            f = w, w = null, m.hasOwnProperty(c) ? (w = m[c], w.updatePos(), w.ox = w.x, w.oy = w.y, w.oSize = w.size, w.color = h) : (w = new r(c, d, p, g, h, f), w.pX = d, w.pY = p), w.isVirus = u, w.isAgitated = l, w.nx = d, w.ny = p, w.nSize = g, w.updateCode = e, w.updateTime = n, f && w.setName(f)
	        }
	        for (e = i.getUint32(o, !0), o += 4, a = 0; e > a; a++) c = i.getUint32(o, !0), o += 4, w = m[c], null != w && w.destroy()
	    }

	    function c(i) {
	        var t = 0,
	            n = new DataView(i.data);
	        switch (240 == n.getUint8(t) && (t += 5), n.getUint8(t++)) {
	            case 16:
	                p(n, t);
	                break;
	            case 20:
	                l = [];
	                break;
	            case 32:
	                var e = n.getUint32(t, !0);
	                l.push(e)
	        }
	    }
	    var h = window.WebSocket,
	        u = window.jQuery,
	        m = [],
	        l = [];
	    r.prototype = {
	        id: 0,
	        points: null,
	        pointsAcc: null,
	        name: null,
	        nameCache: null,
	        sizeCache: null,
	        x: 0,
	        y: 0,
	        size: 0,
	        ox: 0,
	        oy: 0,
	        oSize: 0,
	        nx: 0,
	        ny: 0,
	        nSize: 0,
	        updateTime: 0,
	        updateCode: 0,
	        drawTime: 0,
	        destroyed: !1,
	        isVirus: !1,
	        isAgitated: !1,
	        wasSimpleDrawing: !0,
	        destroy: function() {
	            delete m[this.id], id = l.indexOf(this.id), -1 != id && l.splice(id, 1), this.destroyed = !0, n(this.id)
	        },
	        setName: function(i) {
	            this.name = i
	        },
	        updatePos: function() {
	            -1 != l.indexOf(this.id) && (e(this.id) || t(this.id, i(this.id, this.color)), o(this.id, this.nx, this.ny), s(this.nx, this.ny))
	        }
	    }, window.WebSocket = function(i, t) {
	        console.log("Listen"), void 0 === t && (t = []);
	        var n = new h(i, t);
	        d(this, n, "binaryType"), d(this, n, "bufferedAmount"), d(this, n, "extensions"), d(this, n, "protocol"), d(this, n, "readyState"), d(this, n, "url"), this.send = function(i) {
	            return n.send.call(n, i)
	        }, this.close = function(i, t) {
	            return n.close.call(n, i, t)
	        }, this.onopen = function() {}, this.onclose = function() {}, this.onerror = function() {}, this.onmessage = function() {}, n.onopen = function(i) {
	            return this.onopen.call(n, i)
	        }.bind(this), n.onmessage = function(i) {
	            return c(i), this.onmessage.call(n, i)
	        }.bind(this), n.onclose = function(i) {
	            return this.onclose.call(n, i)
	        }.bind(this), n.onerror = function(i) {
	            return this.onerror.call(n, i)
	        }.bind(this)
	    }, window.WebSocket.prototype = h, a()
	}();

**Comment by Logan7432** [comment]
Thats my minimap code

**Comment by Logan7432** [comment]
Hello?

**Comment by MastaCoder** [comment]
It's me.

**Comment by Logan7432** [comment]
@MastaCoder Could you fix this code so it adjusts to any size minimap?

**Comment by MastaCoder** [comment]
I'll fix it if you can describe what this function is:

```javascript
function p(i, t) {
        var n = +new Date,
            e = Math.random(),
            o = t,
            s = i.getUint16(o, !0);
        o += 2;
        for (var a = 0; s > a; ++a) {
            var d = m[i.getUint32(o, !0)],
                p = m[i.getUint32(o + 4, !0)],
                o = o + 8;
            d && p && (p.destroy(), p.ox = p.x, p.oy = p.y, p.oSize = p.size, p.nx = d.x, p.ny = d.y, p.nSize = p.size, p.updateTime = n)
        }
        for (a = 0;;) {
            var c = i.getUint32(o, !0);
            if (o += 4, 0 == c) break;
            ++a;
            var d = i.getInt16(o, !0),
                o = o + 2,
                p = i.getInt16(o, !0),
                o = o + 2;
            g = i.getInt16(o, !0), o += 2;
            for (var h = i.getUint8(o++), u = i.getUint8(o++), l = i.getUint8(o++), h = (h << 16 | u << 8 | l).toString(16); 6 > h.length;) h = "0" + h;
            var h = "#" + h,
                w = i.getUint8(o++),
                u = !!(1 & w),
                l = !!(16 & w);
            2 & w && (o += 4), 4 & w && (o += 8), 8 & w && (o += 16);
            for (var f, w = ""; f = i.getUint16(o, !0), o += 2, 0 != f;) w += String.fromCharCode(f);
            f = w, w = null, m.hasOwnProperty(c) ? (w = m[c], w.updatePos(), w.ox = w.x, w.oy = w.y, w.oSize = w.size, w.color = h) : (w = new r(c, d, p, g, h, f), w.pX = d, w.pY = p), w.isVirus = u, w.isAgitated = l, w.nx = d, w.ny = p, w.nSize = g, w.updateCode = e, w.updateTime = n, f && w.setName(f)
        }
        for (e = i.getUint32(o, !0), o += 4, a = 0; e > a; a++) c = i.getUint32(o, !0), o += 4, w = m[c], null != w && w.destroy()
    }
```

This is MINIFIED CODE. It's not your code, it has nothing to do with MultiOgar nor does it even have to do anything with Cigar. This is a place for issues, not programming help center 101. Also, if I were to help you I wouldn't spoon feed everything for you. It's best to try try try until you learn.

**Comment by Logan7432** [comment]
@Mastacoder Your correct, it is not my code. I'm very very new to coding. I'm only decently good at html coding. I'm trying my absolute best to try and learn. Since I don't know how to code this advanced yet that's why I'm asking for your help. If you can just help me this last time I'll never bother anyone on here again. Thanks.

**Comment by Logan7432** [comment]
@mastacoder Atleast please walk me through steps so I can get this fixed

**Comment by Logan7432** [comment]
@mastacoder wait is that the code above?

**Comment by Logan7432** [comment]
@MastaCoder Whats that code above for?

**Comment by Logan7432** [comment]
@Mastacoder if you help me with this when I eventually get money from running ads on my site I'll give you like $4. But it will take a week or two.

**Comment by Logan7432** [comment]
In paypal

**Comment by Megabyte918** [comment]
This is a client-related issue.

**Comment by Logan7432** [comment]
come on....

**Comment by Logan7432** [comment]
@MastaCoder I still cant get this to work! Please Help!!!

**Comment by Logan7432** [comment]
@MastaCoder Can you help me yet? I have been messing around with my Main_out.js on my client to get the map to work but no luck... Can you seriously help me out now!?

**Comment by MastaCoder** [comment]
Nop

**Comment by Logan7432** [comment]
@MastaCoder Can you at least tell me which file I would need to navigate to in my client to fix this?

**Comment by MastaCoder** [comment]
Nop

**Comment by Logan7432** [comment]
@MastaCoder Seriously, Why cant you help me. Please.

**Comment by Logan7432** [comment]
@MastaCoder Stop Ignoring me and just help me and Ill stop bugging you for ever

**Comment by Logan7432** [comment]
If someone fixes this for me ill play u

**Comment by MastaCoder** [comment]
Nop
